### PR TITLE
Add devkit

### PR DESCRIPTION
**Pull request by dilin-MS**

### CODE REVIEW

File: src/Models/ArduinoDeviceBase.ts [hunk 1]
```diff
@@ -121,8 +122,54 @@ export abstract class ArduinoDeviceBase implements Device {
...
+
   abstract async create(): Promise<boolean>;
 
+  async createCore(board: Board|undefined, templateFiles: TemplateFileInfo[]):
+      Promise<boolean> {
+    const deviceFolderPath = this.deviceFolder;
+
+    const scaffoldType = ScaffoldType.Local;
+    if (!await FileUtility.directoryExists(scaffoldType, deviceFolderPath)) {
+      throw new Error('Unable to find the device folder inside the project.');
```
Comment (ArthurMa1978) [inline src/Models/ArduinoDeviceBase.ts]: Internal error, couldn't find the template folder

File: src/Models/ArduinoDeviceBase.ts [hunk 2]
```diff
@@ -121,8 +122,54 @@ export abstract class ArduinoDeviceBase implements Device {
...
+    }
+    if (!board) {
+      throw new Error('Unable to find the board in the config file.');
```
Comment (ArthurMa1978) [inline src/Models/ArduinoDeviceBase.ts]: Invalid / unsupported target platform

File: src/Models/ArduinoDeviceBase.ts (hunk 1 above)
Comment (dilin-MS) [reply src/Models/ArduinoDeviceBase.ts]: Fix.

File: src/Models/ArduinoDeviceBase.ts (hunk 2 above)
Comment (dilin-MS) [reply src/Models/ArduinoDeviceBase.ts]: Fix.

File: src/Models/ArduinoDeviceBase.ts [hunk 3]
```diff
@@ -121,8 +122,54 @@ export abstract class ArduinoDeviceBase implements Device {
...
+        scaffoldType, this.deviceFolder);
+
+    for (const fileInfo of templateFiles) {
+      if ((fileInfo.fileName.endsWith('macos.json') ||
+           fileInfo.fileName.endsWith('win32.json'))) {
+        if (fileInfo.fileName.endsWith('macos.json') && (plat === 'darwin')) {
+          await this.generateCppPropertiesFile(scaffoldType, board, fileInfo);
+        } else if (
+            fileInfo.fileName.endsWith('win32.json') && (plat === 'win32')) {
+          await this.generateCppPropertiesFile(scaffoldType, board, fileInfo);
```
Comment (ArthurMa1978) [inline src/Models/ArduinoDeviceBase.ts]: what's the different between drawin and win32 here? both invoke 'await this.generateCppPropertiesFile(scaffoldType, board, fileInfo);'

File: src/Models/ArduinoDeviceBase.ts [hunk 4]
```diff
@@ -121,8 +122,54 @@ export abstract class ArduinoDeviceBase implements Device {
...
+        }
+      } else {
+        // Copy file directly
```
Comment (ArthurMa1978) [inline src/Models/ArduinoDeviceBase.ts]: Linux right?

File: src/Models/ArduinoDeviceBase.ts (hunk 3 above)
Comment (dilin-MS) [reply src/Models/ArduinoDeviceBase.ts]: Fixed.

**Pull request merged**